components/NftCard.tsx

```diff
@@ -1,24 +1,38 @@
 import { useConnection } from '@solana/wallet-adapter-react'
-import { Link, Button, Text, IconButton } from '@primer/react'
+import { Link, Text, Button, Spinner } from '@primer/react'
 import { FC } from 'react'
 import { LinkExternalIcon } from '@primer/octicons-react'
+import { VariantType } from '@primer/react/lib/Button/types'
 
-const NftCard: FC<{ title: string; imageLink: string; hash?: string }> = ({ title, imageLink, hash }) => {
+const NftCard: FC<{
+    title: string
+    imageLink?: string
+    hash?: string
+    buttonProps?: { text: string; isLoading: boolean; variant: VariantType; onClick: () => void }
+}> = ({ title, imageLink, hash, buttonProps }) => {
     const { connection } = useConnection()
     return (
         <div
-            className='d-flex flex-column flex-justify-center flex-items-center border rounded-3 gap-2 py-2'
+            className='d-flex flex-column flex-justify-center flex-items-center border rounded-3 py-2'
             style={{ width: '185px' }}
         >
             <Text as='p' fontWeight='bold'>
                 {title}
             </Text>
             <div
-                className='border rounded-3 mb-2 overflow-hidden d-flex flex-items-center flex-justify-center'
+                className='border color-bg-inset rounded-3 mb-2 overflow-hidden d-flex flex-items-center flex-justify-center'
                 style={{ height: '168px', width: '168px' }}
             >
-                <img alt={title} src={imageLink} />
+                {imageLink && <img alt={title} src={imageLink} />}
             </div>
+            {buttonProps && (
```

Comment:
Instead of this, it could be `{!footer ? '' : footer}`  because sometimes more than 1 button is needed